### PR TITLE
`tests/ui`: A New Order [28/28] FINAL PART

### PR DESCRIPTION
**Pull request by Kivooeo**
> [!NOTE]
>
> Intermediate commits are intended to help review, but will be squashed prior to merge.

Some `tests/ui/` housekeeping, to trim down number of tests directly under `tests/ui/`. Part of rust-lang/rust#133895.

r? @tgross35

### CODE REVIEW

File: tests/ui/binding/underscore-prefixed-function-argument.rs [hunk 1]
```diff
@@ -0,0 +1,15 @@
+//! Test that functions with unnamed arguments are correct handled by compiler
```
Comment (tgross35) [inline tests/ui/binding/underscore-prefixed-function-argument.rs]:
```suggestion
//! Test that argument names starting with `_` are usable.
```

Since it's not actually unnamed

File: tests/ui/modules/module-use-nested-groups.rs [hunk 2]
```diff
@@ -15,10 +17,13 @@ mod a {
 }
 
 // Test every possible part of the syntax
-use a::{B, d::{self, *, g::H}};
-
 // Test a more common use case
-use std::sync::{Arc, atomic::{AtomicBool, Ordering}};
```
Comment (tgross35) [inline tests/ui/modules/module-use-nested-groups.rs]: I think this test is intended to test the nested syntax, so these should probably stay as-is with `#[rustfmt::skip]`. See its addition in 91ba8b42fcce0491fa8a1b95e4088bcabf9abf4a.

File: tests/ui/type/unit-type-basic-usages.rs [hunk 3]
```diff
@@ -0,0 +1,14 @@
+// Checks the basic usage of unit type
```
Comment (tgross35) [inline tests/ui/type/unit-type-basic-usages.rs]:
```suggestion
//! Checks the basic usage of unit type
```

File: tests/ui/hashmap/hashset-enum-variant.rs [hunk 4]
```diff
@@ -1,26 +1,28 @@
+//! Checks that `HashSet` initialization with enum variants correctly includes only
+//! specified variants, preventing platform-specific bugs
+//! where all enum variants were mistakenly included
```
Comment (tgross35) [inline tests/ui/hashmap/hashset-enum-variant.rs]: It's a tricky case to describe. Maybe:

```rust
//! Check for correct initialization of `HashSet` with enums. This is a regression test for a
//! codegen bug that caused the `HashSet` to appear as if it contained one of each enum variant.
```

**Review by tgross35 — COMMENTED** [review]
Looks pretty good, thanks for keeping the summaries simpler here. Just a few comments

**Review by tgross35 — APPROVED** [review]

**Comment by tgross35** [comment]
🎉 congrats on reaching the end!
@bors r+ rollup

**Comment by Kivooeo** [comment]
Thanks for being with me on this journey. However, statistically speaking, this is far from “the end.” 

(I might have forgotten a few tests here and there, but I’d really love to see a clean directory before starting a new issues directory)

But anyway you’ve certainly earned yourself a well-deserved rest… for now

**Comment by tgross35** [comment]
> But anyway you’ve certainly earned yourself a well-deserved rest… for now

Very much the same to you, enjoy :)

File: tests/ui/expr/syntax-edge-cases-lint-clean.rs
Comment (bjorn3) [inline tests/ui/expr/syntax-edge-cases-lint-clean.rs:1]: weird-exprs.rs has been referenced a fair bit in blogs and casual conversations. This move breaks all those references and is a lot harder to remember IMO..

Comment (Kivooeo) [reply tests/ui/expr/syntax-edge-cases-lint-clean.rs:1]: Yeah, it's pretty iconic test, can't argue with that, I guess I can revert the name, is that sounds fine?

Comment (tgross35) [reply tests/ui/expr/syntax-edge-cases-lint-clean.rs:1]: I completely overlooked this rename, yes please do!
